src/network/netdev/xfrm.c

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#include "netdev/xfrm.h"
+
+static int xfrm_config_verify(NetDev *netdev, const char *filename) {
+        Xfrm *x;
+
+        assert(netdev);
+        assert(filename);
+
+        x = XFRM(netdev);
+
+        if (!x->dev) {
+                log_warning("Xfrm NetDev without dev set in %s. Defaulting to 'lo'", filename);
+                if(!((x->dev = strdup("lo")))) {
+                        log_error("Could not allocate memory");
+                        return -ENOMEM;
+                }
+        }
+
+        if (!x->if_id) {
+                log_warning("Xfrm NetDev without if_id set in %s. Defaulting to '0'", filename);
+                x->if_id = 0;
+        }
+
+        return 0;
+}
+
+static int xfrm_fill_message_create(NetDev *netdev, Link *link, sd_netlink_message *message) {
+        int r;
+        int if_idx;
+
+        assert(netdev);
+        assert(message);
+
+        Xfrm *x = XFRM(netdev);
+
+        if_idx = if_nametoindex(x->dev);
+        if (if_idx == 0)
+                return log_netdev_error_errno(netdev, SYNTHETIC_ERRNO(ENODEV), "Could not get interface index, error %m");
+
+        r = sd_netlink_message_append_u32(message, IFLA_XFRM_LINK, if_idx);
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r, "Could not append IFLA_XFRM_LINK, error %m");
+
+        r = sd_netlink_message_append_u32(message, IFLA_XFRM_IF_ID, x->if_id);
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r, "Could not append IFLA_XFRM_IF_ID, error %m");
+
+        return 0;
+}
+
+static void xfrm_done(NetDev *n) {
+        Xfrm *x;
+
+        assert(n);
+
+        x = XFRM(n);
+        x->dev = mfree(x->dev);
+}
+
+const NetDevVTable xfrm_vtable = {
+        .object_size = sizeof(Xfrm),
+        .sections = "Match\0NetDev\0Xfrm\0",
+        .config_verify = xfrm_config_verify,
+        .fill_message_create = xfrm_fill_message_create,
+        .done = xfrm_done,
+        .create_type = NETDEV_CREATE_INDEPENDENT
```

Comment:
I guess, this may introduce race. That is, underlying netdev may not be created yet when creating the xfrm device.